.github/workflows/go.yml

```diff
@@ -54,4 +60,63 @@ jobs:
       run: |
         cd $GOPATH/src/github.com/${{ github.repository }}
         ./test/install_bitcoind.sh
-        ./test/csn_bridge.sh "./cmd/utreexoserver/utreexoserver" "./cmd/utreexoclient/utreexoclient"
+        ./test/csn_bridge.sh "./cmd/utreexoserver/utreexoserver" "./cmd/utreexoclient/utreexoclient" "${{ github.workspace }}/testdatapath"
+
+    - name: Upload testdata
+      uses: actions/upload-artifact@v2
+      with:
+        name: testdatapath
+        path: "${{ github.workspace }}/testdatapath"
+
+  check-proof-backwards:
```

Comment:
It looks a bit weird that there are so many duplicate steps, like "setup env", "Install Go", "Get deps" and "Build" are the same as in the previous job. 
Could this not just be a couple extra steps in the first job?
After "Run integration tests" add: "Checkout master", "Build", "Check backwards".

Reply:
Jobs don't share state so they would have to be uploaded to github artifacts.

I think it'd be more maintainable/functional if the second job fetches and builds itself. That process is fairly quick as well